KhivaCsharp/KhivaCsharp/Matrix.cs

```diff
@@ -0,0 +1,135 @@
+                IntPtr motif_distances; IntPtr motif_indices; IntPtr subsequence_indices;
+                interop.DLLMatrix.find_best_n_motifs(ref profileReference, ref indexReference, ref m, ref n,
+                                                        out motif_distances, out motif_indices, out subsequence_indices,
+                                                        ref self_join);
+                profile.Reference = profileReference;
+                index.Reference = indexReference;
+                var tuple = Tuple.Create(KhivaArray.Create(motif_distances),
+                                         KhivaArray.Create(motif_indices),
+                                         KhivaArray.Create(subsequence_indices));
+                return tuple;
+            }
+
+            /// <summary>
+            /// Primitive of the STOMP algorithm.
+            /// 
+            /// [1] Yan Zhu, Zachary Zimmerman, Nader Shakibay Senobari, Chin-Chia Michael Yeh, Gareth Funning, Abdullah Mueen,
+            /// Philip Brisk and Eamonn Keogh (2016). Matrix Profile II: Exploiting a Novel Algorithm and GPUs to break the one
+            /// Hundred Million Barrier for Time Series Motifs and Joins. IEEE ICDM 2016.
+            /// </summary>
+            /// <param name="tssa">Query time series</param>
+            /// <param name="tssb">Reference time series</param>
+            /// <param name="m">Pointer to a long with the length of the subsequence.</param>
+            /// <returns>Tuple with 
+            /// the matrix profile, whichlects the distance to the closer element of the subsequence from 'tssa' in 'tssb' and 
+            /// the matrix profile index, which points to where the aforementioned minimum is located.</returns>
+            public static Tuple<KhivaArray, KhivaArray> Stomp(KhivaArray tssa, KhivaArray tssb, long m)
+            {
+                IntPtr aReference = tssa.Reference;
+                IntPtr bReference = tssb.Reference;
+                IntPtr p; IntPtr i;
+                interop.DLLMatrix.stomp(ref aReference, ref bReference, ref m,
+                                        out p, out i);
+                tssa.Reference = aReference;
+                tssb.Reference = bReference;
+                var tuple = Tuple.Create(KhivaArray.Create(p),
+                                        KhivaArray.Create(i));
+                return tuple;
+            }
+
+            /// <summary>
+            /// Primitive of the STOMP self join algorithm.
+            /// 
+            /// [1] Yan Zhu, Zachary Zimmerman, Nader Shakibay Senobari, Chin-Chia Michael Yeh, Gareth Funning, Abdullah Mueen,
+            /// Philip Brisk and Eamonn Keogh (2016). Matrix Profile II: Exploiting a Novel Algorithm and GPUs to break the one
+            /// Hundred Million Barrier for Time Series Motifs and Joins. IEEE ICDM 2016.
+            /// </summary>
+            /// <param name="tss">Query anderence time series</param>
```

Comment:
What do you mean with this? <param name="tss">Query anderence time series</param>
By the way missing point at the end